### PR TITLE
Ht 7715

### PR DESCRIPTION
**Pull request by Callumccr**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -1,106 +1,121 @@
-#Module      : LABEL
-#Description : Terraform label module variables.
-variable "name" {
+variable "endpoint_type" {
   type        = string
-  default     = ""
-  description = "Name  (e.g. `app` or `cluster`)."
+  default     = "VPC"
+  description = "Type of SFTP server. **Valid values:** `PUBLIC`, `VPC` or `VPC_ENDPOINT`"
```
Comment (DesmondH0) [inline variables.tf:4]: Can we have validation here as only accepting specific values?

File: variables.tf [hunk 2]
```diff
@@ -1,106 +1,121 @@
...
 }
 
-variable "repository" {
-  type        = string
-  default     = "https://github.com/clouddrove/terraform-aws-sftp"
-  description = "Terraform current module repo"
+variable "protocols" {
+  type        = list(string)
+  default     = ["SFTP"]
+  description = "List of file transfer protocol(s) over which your FTP client can connect to your server endpoint. **Possible Values:** FTP, FTPS and SFTP"
```
Comment (DesmondH0) [inline variables.tf:10]: same here, could have validation

File: variables.tf [hunk 3]
```diff
@@ -1,106 +1,121 @@
...
 }
 
-variable "attributes" {
-  type        = list(any)
-  default     = []
-  description = "Additional attributes (e.g. `1`)."
+variable "identity_provider_type" {
+  type        = string
+  default     = "SERVICE_MANAGED"
+  description = "Mode of authentication to use for accessing the service. **Valid Values:** `SERVICE_MANAGED`, `API_GATEWAY`, `AWS_DIRECTORY_SERVICE` or `AWS_LAMBDA`"
```
Comment (DesmondH0) [inline variables.tf:43]: ditto for validation

**Comment by DesmondH0** [comment]
As using `Optional` from variables, should we include the generated `version.tf` and `humn.tf` in this module as well?
like the Drafted PR? [[ref](https://github.com/humn-ai/tf-humn-iac-live/pull/1201/files)]

Otherwise this module could not work if calling it alone, and we can generate less stuffs from our iac-live repo 🤔

File: variables.tf (hunk 3 above)
Comment (Callumccr) [reply variables.tf]: please check changes

File: variables.tf (hunk 2 above)
Comment (Callumccr) [reply variables.tf]: please check changes

File: variables.tf (hunk 1 above)
Comment (Callumccr) [reply variables.tf]: please check changes

**Comment by Callumccr** [comment]
> As using `Optional` from variables, should we include the generated `version.tf` and `humn.tf` in this module as well? like the Drafted PR? [[ref](https://github.com/humn-ai/tf-humn-iac-live/pull/1201/files)]
> 
> Otherwise this module could not work if calling it alone, and we can generate less stuffs from our iac-live repo 🤔

Good shout, please check changes :)

**Pull request merged**